astropy/io/ascii/cds.py

```diff
@@ -647,6 +650,32 @@ def write(self, lines):
 
                 to_pop.append(i)   # Delete original ``SkyCoord`` column.
 
+            # If the whole column is a `Time` or if it contains `Time` values,
+            # convert values to Modified Julian Dates.
+            elif isinstance(col, Time) or isinstance(col[0], Time):
+                # If column already has a description, then append to it.
+                name, description = 'MJD', 'Modified Julian Date'
+                if isinstance(col, Column):
+                    if col.description is not None:
+                        description = col.description + ' (Modified Julian Date)'
+                    if col.name is not None:
+                        name = col.name
+                try:
+                    self.cols[i] = Column([tval.mjd for tval in col],
```

Comment:
I wonder if this should be changed to `tval.to_value('mjd', 'long')` for `longdouble` precision, and go beyond the `format='.12f'`.

Reply:
Let me see that.
However, is having greater precision than 12 decimal digits is a requirement?

Reply:
Good question. Unfortunately, I don't think we can have a one-size-fits-all answer. Imaging you make a long-term lightcurve over years for a variable star, going back from historical records in 1894 to today. In that case, precision to a day is enough and a precision of 12 digits would in fact be misleading. For the historical records, you won't know the time of the observation better than an hour (or a minute, if they were very good in writing down their notes and all notes survived to today), even for modern observations, the exposure time is often a few min to an hour for faint objects, so there is really no point to be more precise than a minute or so. Everything above that is misleading. On the other hand, the time column could hold the data for pulsar timing, where we want to measure the spin-up or spin-down of the pulsar. Typical pulse rates are 300 Hz or more, so if we want to measure spin-down, we need to be precise on the mirco or nano-second level.

I'm in fact not sure how to address that, since the "time" does not know about the precision needed by the user. Internally, it's just a variable with two floats, but for printing out for human consumption (which is what we do here, even if the format is called machine-readable), we do need that extra bit of information to make sense. For "normal" columns, the user can set the "format" to decide how many digits they want. I don't think time columns have that, though? Maybe @taldcroft has an idea how to address that?

I don't know much about the mix-in columns and don't have time to dig into the code right now to see if this information can be attached by the user in some way, but on top of my head I think we might need to use the "formats" dictionary for the write function. If no format is given for the time column, we pick some default (I suggest 4 decimal digits - a few second precision) and of someone wants more or less, they can change that. (That would profit from an example in the narrative docs. Can be short, just text like "The default is to print time in days as recommended by the MRT standard [link here] with four decimal digits. The precision can be changed like this: ``tab.write(..., formats={'name of time colum'}: '%15.12f')``." - no need to have an example table and an executable example here, just a line of inline text woudl be enough, but we may need a test for this.)